source/reference/api/groovy-api.rst

```diff
@@ -69,6 +69,62 @@ date is set as the attribute. Assume that the REST script exists under Scripts >
 
     return ["date": date]
 
+""""""""""""""
+Path Variables
+""""""""""""""
+Path variables (``pathVars``) allows you to pass a value as part of the URL. It's a part of the path of the document
+to be accessed during the API call. In a REST controller the system can automatically pick out those portions of
+the URL and feed them to you with the name you supplied.
+
+To specify ``pathVars`` for a REST API, simply name the folder as the variables, for example, for the following API
+call ``http://mysite/api/1/services//bar/{model}/{make}/{year}/foo``, to create the pathVars ``{model}``,
+``{make}`` and ``{year}``, your folder structure under ``/scripts/rest`` should look like this: ``/bar/{model}/{make}/{year}/foo``.
+
+Let's take a look at an example of creating a REST API ``http://mysite/api/1/services/foo/{version}/helloworld.json``
+where, ``{version}`` could either be ``1`` or ``2``.
+
+First, we'll set up the folder structure for the API. Under ``scripts`` > ``rest``, add a folder named ``foo``, then
```

Comment:
This example is wrong. The whole point of `pathVars` is that you just need a single script instead of one per possible value in the request. In the paragraphs above you're even confirming this by saying that you need to create a folder structure with folders that have `{}` in their name, like `{model}`.

So if we want to use the example, you would need to create the following folders: `scripts` > `rest` > `foo` > `{value}` > `helloworld.groovy`. The script then would have this content:

```
def value = pathVars.value
return 'Hello world version ${value}!'
```

And if you call http://mysite/api/1/services/foo/1/helloworld.json, then you should get `Hello world version 1`.

Please make sure you test this locally first.

Reply:
Btw I think we need to add that `pathVars` is a variable available to Groovy scripts: https://docs.craftercms.org/en/4.1/reference/api/groovy-api.html. And just like `params`, `headers` and `cookies` it's a `Map`.